### ISSUE TITLE
Inside the after call back record is null when creating resource in modal mode

### ISSUE THREAD

**Issue by alepieser** [laravel-filament/filament#5844]
### Package

filament/filament

### Package Version

v2.17.11

### Laravel Version

v10.2.0

### Livewire Version

v2.12.2

### PHP Version

8.2.3 

### Problem description

Inside the `after` call back after inserting a record in the database, the $record  is null

### Expected behavior

To have access to the record just created

### Steps to reproduce

```php
class ManageAuthors extends ManageRecords
{
    protected static string $resource = AuthorResource::class;

    protected function getActions(): array
    {
        return [
            Actions\CreateAction::make()
                ->after(function ($record) {
                    dd($record);
                }),
        ];
    }
}
```

### Reproduction repository

https://github.com/alepieser/filament_bug

**Comment by zepfietje** [comment]
Have you tried `function (Model $record)`?

**Comment by alepieser** [comment]
> Model $record

Adding the type, I got this error:

`App\Filament\Resources\Blog\AuthorResource\Pages\ManageAuthors::App\Filament\Resources\Blog\AuthorResource\Pages\{closure}(): Argument #1 ($record) must be of type Illuminate\Database\Eloquent\Model, null given`

**Comment by Z3d0X** [comment]
Potentially relevant commit: https://github.com/filamentphp/filament/commit/0de95f1882f368ae6d6a0712dde33facd5f4ed8c

It seems like record is set back to null in `->action()`, therefore it is null in `->after()` hook. Commenting this line seems to resolve the record properly, but i assume it was placed there to fix some other bug?

**Comment by danharrin** [comment]
That was used to fix #3364

**Comment by ricventu** [comment]
Same issue here.

adding action to a ViewPage:

```php
    protected function getActions(): array
    {
        return [
            Actions\Action::make('my_action')
            ->hidden(static function(Model $record) {
                return false;
            }),
       ]
   }
```

Package Version
v2.17.16

Laravel Version
v9.52.4

PHP Version
8.1.16

**Comment by danharrin** [comment]
Fixed for CreateAction in e9797d395d363ad1b8560f5faa76b8135a16002c.

@ricventu your issue is different. The `$record` isn't automatically passed to actions on the view page, only DeleteAction and EditAction. You just need to pass in `->record($this->record)` or use `$this->record` directly in the action

**Comment by ricventu** [comment]
> Fixed for CreateAction in [e9797d3](https://github.com/filamentphp/filament/commit/e9797d395d363ad1b8560f5faa76b8135a16002c).
> 
> @ricventu your issue is different. The `$record` isn't automatically passed to actions on the view page, only DeleteAction and EditAction. You just need to pass in `->record($this->record)` or use `$this->record` directly in the action

Thanks!

**Comment by Vsonigit** [comment]
@danharrin in my Relationmanager for a resource:
Forms\Components\Select::make('project_id')
                    ->label('Select Project')
                    ->options(fn(callable $get) => $get('record')->projects->pluck('title', 'id')->toArray())
                    ->hidden(function ($record) {
                        Debugbar::info('get', ($record));
                    }) the $record is returning null and it happening for several cases, can you give the correct direction or provide a solution. Thanks!!!
